torch/csrc/distributed/rpc/init.cpp

```diff
@@ -181,15 +181,17 @@ PyObject* rpc_init(PyObject* /* unused */) {
                   If the current node is the owner, returns a reference to the
                   local value. Otherwise, throws an exception.
               )")
-          .def(py::pickle(
-              [](const PyRRef& self) {
-                // __getstate__
-                return self.pickle();
-              },
-              [](py::tuple t) { // NOLINT
-                // __setstate__
-                return PyRRef::unpickle(t);
-              }))
+          .def(
+              py::pickle(
+                  [](const PyRRef& self) {
+                    // __getstate__
+                    return self.pickle();
+                  },
+                  [](py::tuple t) { // NOLINT
+                    // __setstate__
+                    return PyRRef::unpickle(t);
+                  }),
+              py::call_guard<py::gil_scoped_release>())
```

Comment:
would I be correct if I assume this GIL acquirement is moved to `rref_impl.cpp`?

Reply:
yes, but it could wait for pythonrpchandler to be constructed in another thread, and that thread is trying to aquire GIL, if we do not release GIL in unpickle, it could result in deadlock